contracts/src/core/BatcherPaymentService.sol

```diff
@@ -220,9 +261,13 @@ contract BatcherPaymentService is
         }
 
         if (leaves.length == 1) {
-            require(leaves[0] == batchMerkleRoot, "Invalid merkle root");
+            if (leaves[0] != batchMerkleRoot) {
+                revert InvalidMerkleRoot(batchMerkleRoot, leaves[0]);
+            }
         } else {
-            require(layer[0] == batchMerkleRoot, "Invalid merkle root");
+            if (layer[0] != batchMerkleRoot) {
+                revert InvalidMerkleRoot(batchMerkleRoot, layer[0]);
+            }
```

Comment:
Change to else if instead of nested

Reply:
Solved in [c9b32e9](https://github.com/yetanotherco/aligned_layer/pull/843/commits/c9b32e95f5fd91f19c54dfb4ed6d97655e7da8d5)